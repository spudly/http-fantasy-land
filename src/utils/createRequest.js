import deepFreeze from 'deep-freeze';

const createRequest = nativeRequest =>
  deepFreeze({
    method: nativeRequest.method,
    url: nativeRequest.url,
    headers: {
      ...nativeRequest.headers,
    },
  });

export default createRequest;
