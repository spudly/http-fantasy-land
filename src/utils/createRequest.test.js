import createRequest from './createRequest';

test('creates a request object from the passed http request object', () => {
  const nativeRequest = {
    method: 'POST',
    url: '/path/to/something',
    headers: {
      Accept: 'text/css',
    },
  };

  expect(createRequest(nativeRequest)).toMatchSnapshot();
});
