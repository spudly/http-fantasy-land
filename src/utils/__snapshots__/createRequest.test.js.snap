// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a request object from the passed http request object 1`] = `
Object {
  "headers": Object {
    "Accept": "text/css",
  },
  "method": "POST",
  "url": "/path/to/something",
}
`;
